### ISSUE TITLE
Support "sls deploy function"

### ISSUE THREAD

**Issue by koterpillar** [seek-oss/serverless-haskell#20]
Possibly useful: https://github.com/DieProduktMacher/serverless-env-generator/blob/develop/src/index.js#L34

**Comment by NickSeagull** [comment]
Any ideas on how to start working on this?

**Comment by koterpillar** [comment]
One would need to check what hooks are called when `deploy function` runs, and hook up the existing code to build the handlers (see hooks added for `invoke local` as an example).

The example project can be a good testing ground, and there is an integration test that can be expanded to include this.

**Comment by NickSeagull** [comment]
So the idea would be to have one executable field in the cabal/package.yaml file per function right?

**Comment by koterpillar** [comment]
That's already what's happening anyway - you have to specify, for each function, which executable needs to be called.

Did you try `deploy function` as is?

**Comment by NickSeagull** [comment]
I'm not sure if I understood that last comment properly, and I think it is becaus I didn't ask the question properly :sweat_smile: Let me rephrase it:

What is the goal with supporting `deploy function`? To support deploying single Haskell functions, or to support having multiple executables per project?

**Comment by koterpillar** [comment]
You can already have multiple executables and multiple lambdas deployed from the same project.

The idea of `deploy function` is to skip generating and updating the CloudFormation stack and directly updating the function code via a single AWS call. This is faster than `deploy`, but messes up CF so isn't recommended for production: https://serverless.com/framework/docs/providers/aws/cli-reference/deploy-function/

I'm wondering exactly what it tried to deploy then, maybe the whole directory including `.stack-work`? That would be an indication that no handlers in the plugin ran, because it sets up the exclude patterns properly and only the actual executable gets uploaded.

**Comment by NickSeagull** [comment]
Great, I wasn't aware how that worked, my fault.

I'm gonna research this further and see what happened to my deploy. Thanks!
